sdk/spring-sdk/src/main/java/kalix/springsdk/annotations/Acl.java

```diff
@@ -18,26 +18,75 @@
 
 import java.lang.annotation.*;
 
+
+/**
+ * Defines ACL configuration for a resource.
+ */
 @Target({ElementType.TYPE, ElementType.METHOD})
 @Retention(RetentionPolicy.RUNTIME)
 @Documented
 public @interface Acl {
 
+  /**
+   * Principals that are allowed to access this resource.
+   * An incoming request must have at least one principal associated with it in this list to be allowed.
+   */
   Matcher[] allow() default {};
 
+
+  /**
+   * After matching an allow rule, an incoming request that has at least one principal
+   * that matches a deny rule will be denied.
+   */
   Matcher[] deny() default {};
 
+
+  /**
+   * The gRPC status code to respond with when access is denied.
+   *
+   * By default, this will be 7 (permission denied), but alternatives might include 16 (unauthenticated) or 5 (not
+   * found). If 0, indicates that the code should be inherited from the parent (regardless of the inherit field).
+   *
+   * When HTTP transcoding is in use, this code will be translated to an equivalent HTTP status code.
+   */
```

Comment:
This is something that I didn't realise before. The deny code are gRPC codes. I think we can leave it as it for now, but we may want to change it to http status code and then convert to proto under the hood.